### DISCUSSION TITLE
Could you clarify the difference between enabled and disabled single hit option?

### DISCUSSION THREAD

**Discussion by voroninp** [coverlet-coverage/coverlet#1405] [Q&A] [answered]
Here's the description:

> SingleHit. Specifies whether to limit code coverage hit reporting to a single hit for each location.

I am not getting how multiple hits can change the coverage statistics, could you give me an example, please? And does this option have any performance implications?

**Comment by petli** [comment, marked as answer]
This is intended as a performance improvement for code that iterates many, many times over some code blocks and the exact number of times isn't important. The coverage statistics in terms of covered lines and branches taken is not affected by this option.

**Reply by voroninp** [reply]
Why would you want it turned off then?

**Reply by petli** [reply]
For performance reasons, when counting every hit takes too much time in aggregate for the test. If you haven't had that problem, you don't need this option either.

**Reply by voroninp** [reply]
I understand. But what is the benefit of counting every hit? What does it give me, if it does not affect coverage?

**Reply by petli** [reply]
Well, the obvious thing is finding out if you hit code as many times as you expect. But that is up to you if you have need for it or not. Counting every hit was the original behaviour for coverlet with this option added later, hence the default is to keep counting every hit to not add a breaking change.

**Reply by voroninp** [reply]
Thanks. Now it's clear. We have a code which contains multiple nested loops and intensive calculations, so every line is hit thousand times. And when we are using dotCover whole CI cycle takes 4 times longer. I expect Coverlet to be fast with this option ;-)

**Reply by voroninp** [reply]
@petli , I've checked it, and as I can see it's incomparably fast! ;-)

**Comment by petli** [comment]
Ok, then this would benefit you. It would be interesting to find out what runtimes you get with coverlet compared to dotCover!
